pkg/backend/openshiftcluster.go

```diff
@@ -179,18 +172,27 @@ func (ocb *openShiftClusterBackend) heartbeat(ctx context.Context, cancel contex
 	}
 }
 
-func (ocb *openShiftClusterBackend) updateAsyncOperation(ctx context.Context, id string, oc *api.OpenShiftCluster, provisioningState, failedProvisioningState api.ProvisioningState) error {
+func (ocb *openShiftClusterBackend) updateAsyncOperation(ctx context.Context, id string, oc *api.OpenShiftCluster, provisioningState, failedProvisioningState api.ProvisioningState, backendErr error) error {
 	if id != "" {
+		log := ocb.baseLog.WithField("clusterID", oc.ID)
 		_, err := ocb.db.AsyncOperations.Patch(ctx, id, func(asyncdoc *api.AsyncOperationDocument) error {
 			asyncdoc.AsyncOperation.ProvisioningState = provisioningState
 
 			now := time.Now()
 			asyncdoc.AsyncOperation.EndTime = &now
 
-			if provisioningState == api.ProvisioningStateFailed {
-				asyncdoc.AsyncOperation.Error = &api.CloudErrorBody{
-					Code:    api.CloudErrorCodeInternalServerError,
-					Message: "Internal server error.",
+			if provisioningState == api.ProvisioningStateFailed && backendErr != nil {
```

Comment:
in what circumstances can `provisioningState == api.ProvisioningStateFailed && backendErr == nil`?  Why is the `backendErr != nil` test needed?  Note: if it is nil, below `err, ok := backendErr.(*api.CloudError)` should return `nil, false`, I think?